src/gui/layout/qgslayoutmousehandles.cpp

```diff
@@ -194,6 +194,42 @@ void QgsLayoutMouseHandles::endMacroCommand()
   mLayout->undoStack()->endMacro();
 }
 
+void QgsLayoutMouseHandles::drawMovePreview( QPainter *painter )
+{
+  const QList<QGraphicsItem *> selectedItems = selectedSceneItems( false );
+  if ( selectedItems.isEmpty() )
+  {
+    return;
+  }
+
+  QList<QGraphicsItem *> itemsToDraw;
```

Comment:
The paint result will need to be cached here in a QImage, as the calls to item->paint can potentially be quite expensive. So we should only do this at most once per move operation. (We should also consider flagging the slow to paint items with a new item flag and skip the preview for those...)